### ISSUE TITLE
1.5.16 broke stuff

### ISSUE THREAD

**Issue by PerpSearch** [koillection/koillection#1255]
Apparently there was an update last night and watchtower did what it does and updated my image from 1.5.15 > 1.5.16 which made some access to my records inaccessible.  I get the 'Oops, a critical error occurred :( [Return to homepage]'   The logs are attached 
[_Koillection_logs.txt](https://github.com/user-attachments/files/17997064/_Koillection_logs.txt)
[_Koillection-DB_logs.txt](https://github.com/user-attachments/files/17997065/_Koillection-DB_logs.txt)

**Comment by PerpSearch** [comment]
Now I'm confused. I started from scratch and now it won't even start a new db.  This is the latest log
      

**** 1/11 - Make sure /uploads folders exist ****

**** 2/11 - Create the symbolic link for the /uploads folder ****

**** 3/11 - Setting env variables ****

**** 4/11 - Migrate the database ****

[2024-12-03T21:30:31.852850+00:00] deprecation.INFO: Deprecated: Lexik\Bundle\JWTAuthenticationBundle\Services\JWSProvider\LcobucciJWSProvider::__construct(): Implicitly marking parameter $clock as nullable is deprecated, the explicit nullable type must be used instead {"exception":"[object] (ErrorException(code: 0): Deprecated: Lexik\\Bundle\\JWTAuthenticationBundle\\Services\\JWSProvider\\LcobucciJWSProvider::__construct(): Implicitly marking parameter $clock as nullable is deprecated, the explicit nullable type must be used instead at /var/www/koillection/vendor/lexik/jwt-authentication-bundle/Services/JWSProvider/LcobucciJWSProvider.php:47)"} []

[2024-12-03T21:30:31.853893+00:00] deprecation.INFO: Deprecated: Lexik\Bundle\JWTAuthenticationBundle\Services\JWTManager::__construct(): Implicitly marking parameter $payloadEnrichment as nullable is deprecated, the explicit nullable type must be used instead {"exception":"[object] (ErrorException(code: 0): Deprecated: Lexik\\Bundle\\JWTAuthenticationBundle\\Services\\JWTManager::__construct(): Implicitly marking parameter $payloadEnrichment as nullable is deprecated, the explicit nullable type must be used instead at /var/www/koillection/vendor/lexik/jwt-authentication-bundle/Services/JWTManager.php:33)"} []

[notice] Migrating up to App\Migrations\Postgresql\Version20241029150732

[notice] Migration App\Migrations\Mysql\Version20200429093452 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20200727114843 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20200812160901 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20200923193955 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20201010121132 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20201215214207 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20210127105342 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20210514100228 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20211207185256 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220206133834 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220206175335 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220208094335 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220630100634 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220701175209 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220713152238 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220716201458 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220717140437 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220722163001 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220722192217 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220725204054 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220829085551 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220909124604 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220916162932 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220917151134 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220918183100 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220920150306 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20220922183351 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20230317131927 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20230425131232 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20230519201736 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20230525100104 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20230613175623 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20230726074848 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20230801111844 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20230801130420 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20230803153034 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20231124105513 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20240105142108 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20240119000437 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20240204172429 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20240205152905 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20240329230249 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20240411083658 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20240906163226 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] Migration App\Migrations\Mysql\Version20241029164924 skipped during Execution. Reason: "Migration can only be executed safely on 'mysql' or 'mariadb'."

[notice] finished in 33290.3ms, used 62.5M memory, 81 migrations executed, 321 sql queries

 [OK] Successfully migrated to version:                                         

      App\Migrations\Postgresql\Version20241029150732                           

**** 5/11 - Create API keys ****

 [OK] Done!                                                                     

**** 6/11 - Refresh caches ****

Refreshing cached values for collections...

Refreshing cached values for wishlists...

Refreshing cached values for albums...

Done!

**** 7/11 - Create user and use PUID/PGID ****

-e  	User UID :	1026

-e  	User GID :	100

**** 8/11 - Set Permissions ****

**** 9/11 - Create nginx log files ****

**** 10/11 - Create symfony log files ****

**** 11/11 - Setup complete, starting the server. ****

and while the containers are running the browser looks like this

Unable to write in the cache directory (/var/www/koillection/var/cache/prod/twig/63).![Screenshot_20241203-171555.png](https://github.com/user-attachments/assets/23c6c0b5-3775-403a-aa50-d6421ccc07e3)

![Screenshot_20241203-171608.png](https://github.com/user-attachments/assets/64e28f7c-6f36-4ea6-9908-edc705fb2c5a)

**Comment by PerpSearch** [comment]
In case you're reading this, I have 1.15.16 running q/ zero data, but as soon as I restore my original data files, it all stops.  You probably have enough to read through so I'll back off for a while until you digest it all.
Thanks for everything
C

**Comment by benjaminjonard** [comment]
This is another issue, there is an access right problem on the cache folder. If I understand correctly you changed your docker compose file for another one, the problem probably comes from here (I see it meddles with GID and UID, I cannot guarantee it works properly).
You should try to use your usual setup and if the problem still persist we can  restart the investigation from here.

**Comment by PerpSearch** [comment]
[koillection.txt](https://github.com/user-attachments/files/18013396/koillection.txt)
[_Koillection_logs.txt](https://github.com/user-attachments/files/18013410/_Koillection_logs.txt)
I got the original docker compose [from Mariushosting](https://mariushosting.com/how-to-install-koillection-on-your-synology-nas/) and when things stopped working the other day I checked his site and he updated it on 12/03 to work with 1.5.16.   

I figured that would make a difference so I deployed the new compose and it didn't help.  As to the GID and UID , I don't know exactly what changed between the two docker compose version, but those values haven't changed in either version. 

After trying 1.5.14 again, it did not work so I'm went back to try 1.5.13 and work w/ the earliest data file from 11/29 and I'm back up.  I really would like to stay current, but after 16 hours of poking at this, I'm too nervous to try at the moment.

I'm attaching my current docker compose file and log, If you have time I'd appreciate any suggestions you have that may allow me to update, but until then I'm going to stay on 13.    Oops, forgot to attach the files.  Thanks again and I look forward to being on the current version whatever and whenever that is.

C
